src/adzerk/figwheel.clj

```diff
@@ -0,0 +1,149 @@
+(ns ^{:doc "Figwheel binding and/or functions copied over"
+      :author "Andrea Richiardi"}
+    adzerk.figwheel
+  (:require [clojure.edn :as edn]
+            [clojure.set :as set]
+            [clojure.java.io :as io]
+            [clojure.string :as str]
+
+            [boot.util :as util]
+            [cljs.build.api :as build-api]
+            [figwheel-sidecar.build-middleware.javascript-reloading :as js-reload]
+            [figwheel-sidecar.cljs-utils.exception-parsing :as cljs-ex]
+            [clojure.tools.namespace.file :as ctn-file]
+            [clojure.tools.namespace.find :as ctn-find]
+            [clojure.tools.namespace.parse :as ctn-parse]
+            [clojure.java.classpath :as cp]
+            [cljs.compiler]
+            [cljs.build.api :as bapi]))
+
+(defn read-msg [data]
+  (try
+    (let [msg (edn/read-string data)]
+      (if (and (map? msg) (:figwheel-event msg)) msg {}))
+    (catch Exception e
+      (util/fail "%s\n" (.getMessage e))
+      (util/print-ex e)
+      {})))
+
+(defn file-namespace
+  "Return the namespace of the input java.io.File"
+  ([^java.io.File file]
+   (file-namespace file nil))
+  ([^java.io.File file read-opts]
+   (-> file
+       (ctn-file/read-file-ns-decl read-opts)
+       (ctn-parse/name-from-ns-decl))))
+
+;; Reusing figwheel's make-sendable-file is not possible at the moment
+;; https://github.com/bhauman/lein-figwheel/blob/e47da1658a716f83888e5a5164ee88e59b2d8c1e/sidecar/src/figwheel_sidecar/build_middleware/notifications.clj#L78
+;; (intern *ns* 'make-sendable-file (var-get #'notifications/make-sendable-file))
+
+(defn build-client-path
+  "Return the path that the client uses (relative to :asset-path)
+
+  The input file map "
+  [compile-opts file-map]
+  (assert (:relative-path file-map) "The file-map is missing some fields, this is a bug.")
+  (-> (str/replace (:relative-path file-map)
+                   (or (:asset-path file-map) "")
+                   "")
+      (str/replace #"^/" "")))
+
+(defn guess-namespace
+  [compile-opts file-map]
+  (let [js-file-path (:full-path file-map)]
+    (assert (string? js-file-path) ":full-path must be a string, This is an bug, please report it.")
+    (assert (re-find #"\.js$" js-file-path) ":full-path must point to a Javascript file. This is an bug, please report it.")
+    (->> js-file-path
+         (js-reload/js-file->namespaces compile-opts)
+         first
+         cljs.compiler/munge)))
```

Comment:
Don't like this myself, I prefer maybe to pass the munged namespace from `boot-cljs`